---
title: "Making NJ Tree for WiDiv Panel"
author: "Patrick Monnahan"
output: html_notebook
---

#Filter for biallelic variants that pass filters and exclude indels
#bcftools view -f PASS --max-alleles 2 --exclude-types indels B73v4_10_Platy.vcf_temp_1.fmt.vcf.gz | bgzip > B73v4_10_Platy.vcf_temp_1.fmt2.vcf.gz

#About 50k random sites from chrom 10 were sufficient to build distance matrix.  Randomly select 50k sites with (use shuf on MSI):
  zgrep -v "#" Merged.vcf.gz | gshuf -n 50000 | bgzip > tmp.vcf.gz
  zgrep "#" Merged.vcf.gz > head.txt
  cat head.txt tmp.vcf.gz | vcf-sort | bgzip > Merged.50k.vcf.gz
  
   
#Load neccessary modules and define functions
```{r setup, include=FALSE}
knitr::opts_chunk$set(include = TRUE)
library(vcfR)
library(adegenet)
library(adegraphics)
library(pegas)
library(StAMPP)
library(lattice)
library(gplots)
library(ape)
library(ggplot2)
library(cowplot)
library(tidyverse)

glPcaFast <- function(x,
 center=TRUE,
 scale=FALSE,
 nf=NULL,
 loadings=TRUE,
 alleleAsUnit=FALSE,
 returnDotProd=FALSE){

 if(!inherits(x, "genlight")) stop("x is not a genlight object")
 # keep the original mean / var code, as it's used further down
 # and has some NA checks..
 if(center) {
 vecMeans <- glMean(x, alleleAsUnit=alleleAsUnit)
 if(any(is.na(vecMeans))) stop("NAs detected in the vector of means")
 }
 if(scale){
 vecVar <- glVar(x, alleleAsUnit=alleleAsUnit)
 if(any(is.na(vecVar))) stop("NAs detected in the vector of variances")
 }
 # convert to full data, try to keep the NA handling as similar
 # to the original as possible
 # - dividing by ploidy keeps the NAs
 mx <- t(sapply(x$gen, as.integer)) / ploidy(x)
 # handle NAs
 NAidx <- which(is.na(mx), arr.ind = T)
 if (center) {
 mx[NAidx] <- vecMeans[NAidx[,2]]
 } else {
 mx[NAidx] <- 0
 }
 # center and scale
 mx <- scale(mx,
 center = if (center) vecMeans else F,
 scale = if (scale) vecVar else F)
 # all dot products at once using underlying BLAS
 # to support thousands of samples, this could be
 # replaced by 'Truncated SVD', but it would require more changes
 # in the code around
 allProd <- tcrossprod(mx) / nInd(x) # assume uniform weights
 ## PERFORM THE ANALYSIS ##
 ## eigenanalysis
 eigRes <- eigen(allProd, symmetric=TRUE, only.values=FALSE)
 rank <- sum(eigRes$values > 1e-12)
 eigRes$values <- eigRes$values[1:rank]
 eigRes$vectors <- eigRes$vectors[, 1:rank, drop=FALSE]
 ## scan nb of axes retained
 if(is.null(nf)){
 barplot(eigRes$values, main="Eigenvalues", col=heat.colors(rank))
 cat("Select the number of axes: ")
 nf <- as.integer(readLines(n = 1))
 }
 ## rescale PCs 
 res <- list()
 res$eig <- eigRes$values
 nf <- min(nf, sum(res$eig>1e-10))
 ##res$matprod <- allProd # for debugging
 ## use: li = XQU = V\Lambda^(1/2)
 eigRes$vectors <- eigRes$vectors * sqrt(nInd(x)) # D-normalize vectors
 res$scores <- sweep(eigRes$vectors[, 1:nf, drop=FALSE],2,
sqrt(eigRes$values[1:nf]), FUN="*")
 ## GET LOADINGS ##
 ## need to decompose X^TDV into a sum of n matrices of dim p*r
 ## but only two such matrices are represented at a time
 if(loadings){
 if(scale) {
 vecSd <- sqrt(vecVar)
 }
 res$loadings <- matrix(0, nrow=nLoc(x), ncol=nf) # create empty matrix
 ## use: c1 = X^TDV
 ## and X^TV = A_1 + ... + A_n
 ## with A_k = X_[k-]^T v[k-]
 myPloidy <- ploidy(x)
 for(k in 1:nInd(x)){
 temp <- as.integer(x@gen[[k]]) / myPloidy[k]
 if(center) {
 temp[is.na(temp)] <- vecMeans[is.na(temp)]
 temp <- temp - vecMeans
 } else {
 temp[is.na(temp)] <- 0
 }
 if(scale){
 temp <- temp/vecSd
 }
 res$loadings <- res$loadings + matrix(temp) %*% eigRes$vectors[k,
1:nf, drop=FALSE]
 }
 res$loadings <- res$loadings / nInd(x) # don't forget the /n of X_tDV
 res$loadings <- sweep(res$loadings, 2, sqrt(eigRes$values[1:nf]),
FUN="/")
 }
 ## FORMAT OUTPUT ##
 colnames(res$scores) <- paste("PC", 1:nf, sep="")
 if(!is.null(indNames(x))){
 rownames(res$scores) <- indNames(x)
 } else {
 rownames(res$scores) <- 1:nInd(x)
 }
 if(!is.null(res$loadings)){
 colnames(res$loadings) <- paste("Axis", 1:nf, sep="")
 if(!is.null(locNames(x)) & !is.null(alleles(x))){
 rownames(res$loadings) <- paste(locNames(x),alleles(x), sep=".")
 } else {
 rownames(res$loadings) <- 1:nLoc(x)
 }
 }
 if(returnDotProd){
 res$dotProd <- allProd
 rownames(res$dotProd) <- colnames(res$dotProd) <- indNames(x)
 }
 res$call <- match.call()
 class(res) <- "glPca"
 return(res)
}

# Return a vector containing color values that vary depending on whether samples satisfy a criteria determined by the stats file
filtColors = function(query, filter_df, filter_column, filter_value){
  cc=c()
  for (i in 1:length(query)){
    if (query[i] %in% as.vector(filter_df[which(filter_df[,filter_column] < filter_value),]$Sample)){
      cc=c(cc,"red")}
      else {
        cc=c(cc,"black")} }
  return(cc)
}

# Filter a list of query sample names by criteria in the stats file
filter = function(query, filter_df, filter_column, filter_value){
  cc=c()
  for (i in 1:length(query)){
    if (query[i] %in% as.vector(filter_df[which(filter_df[,filter_column] > filter_value),]$Sample)){
      cc=c(cc,query[i])}
  }
  return(cc)
}

# Return a vector containing color values that denote species
tipColors = function(query, filter_df){
  cc=c()
  for (i in 1:length(query)){
    if (query[i] %in% as.vector(filter_df$Sample)){
      cc=c(cc,as.character(filter_df[filter_df$Sample == query[i], 'SpecCol']))}
  }
  return(cc)
}

```

#Load data
```{r}
# vcf = read.vcfR("~/Documents/Research/Maize/MaizeSV/data/Merged.5.50k.flt.vcf.gz")
vcf = read.vcfR("~/Documents/Research/Maize/MaizeSV/data/VCFs/WiDiv_11-2019_50k_NJtree.flt.rcdHets.vcf.gz")
Vcf = vcfR2genlight(vcf)
pops = read.table("~/Documents/Research/Maize/widiv/Maize_genotype_groups2.txt", head = T)

locNames(Vcf) <- paste(vcf@fix[,1],vcf@fix[,2],sep="_")

pop(Vcf)<-substr(indNames(Vcf),1,3)

#Generate color key for dupliciate fastqs (red) versus samples with just one fastq (black)
dd = data.frame(Vcf$ind.names)
dd %<>% mutate(ord = row_number(), xx = str_replace(str_replace(Vcf.ind.names, "GH17_", ""), "2:", "")) %>% mutate(V2 = str_split_fixed(xx, "_", 2)[,1]) %>% mutate(names = paste(V2, row_number(), sep = "-"))
dd %<>% mutate(unique = case_when(!V2 %in% dd[duplicated(dd$V2),]$V2 ~ "black", V2 %in% dd[duplicated(dd$V2),]$V2 ~ "red"))

names = read.csv("~/Documents/Research/Maize/MaizeSV/misc/name_key.csv", head = F)
kk = merge(dd, names, by = "V2", all.x = T)
kk %<>% mutate(V1 = as.character(V1)) %>% mutate(Nam = case_when(!is.na(V1) ~ V1, is.na(V1) ~ as.character(V2)))
kk %<>% mutate(Names = paste(Nam, row_number(), sep = "-"))
kk %<>% mutate(unique = case_when(!Nam %in% kk[duplicated(kk$Nam),]$Nam ~ "black", Nam %in% kk[duplicated(kk$Nam),]$Nam ~ "red"))
kk %<>% arrange(ord)

#Format Heterotic group info for coloring tree tips.
grps = read.csv("~/Documents/Research/Maize/widiv/HetGroups.csv")
grps %<>% mutate(Nam=Genotype)
KK = merge(kk, grps, by="Nam", all.x=T)

#Set colors
cols=c("gray48","red","blue","pink4","green","darkolivegreen3","darkolivegreen4","gold","lightblue","turquoise4","wheat3","lightpink","saddlebrown")
names=c(".","SSS","NSS","Flint","Iodent","Iodent/Midland","Midland","Popcorn", "Tropical", "Tropical/NSS", "Sweet Corn", "Semi Flint", "UK")
Cols = data.frame(Cols = cols, Group = names)

KK = merge(KK, Cols, by="Group", all.x=T)
KK %<>% mutate(Cols = case_when(is.na(Cols) ~ "black", TRUE ~ as.character(Cols)))
KK %<>% arrange(ord)
```

### Find samples with high missingness.  Passed a certain level, missingness will crash distance calculations.
```{r}
gt <- extract.gt(vcf, element = "GT", as.numeric=TRUE)
colSums(is.na(gt))

sum((gt$`IBB15` - gt$`r1049_FDPL190729869-1a_HTWF3DSXX_L3_B73v4`)^2, na.rm=T) / sum(!is.na((gt$`IBB15` - gt$`r1049_FDPL190729869-1a_HTWF3DSXX_L3_B73v4`)^2), na.rm=T) * 100
sum((gt$`L-135` - gt$`r1022_FDPL190729849-1a_HTWF3DSXX_L2_B73v4`)^2, na.rm=T) / sum(!is.na((gt$`L-135` - gt$`r1022_FDPL190729849-1a_HTWF3DSXX_L2_B73v4`)^2), na.rm=T) * 100

sum((gt$`L-135` - gt$`IBB15`)^2, na.rm=T) / sum(!is.na((gt$`L-135` - gt$`IBB15`)^2), na.rm=T) * 100
sum((gt$`L-135` - gt$`r1049_FDPL190729869-1a_HTWF3DSXX_L3_B73v4`)^2, na.rm=T) / sum(!is.na((gt$`L-135` - gt$`r1049_FDPL190729869-1a_HTWF3DSXX_L3_B73v4`)^2), na.rm=T) * 100

sum((gt$`r1022_FDPL190729849-1a_HTWF3DSXX_L2_B73v4` - gt$`IBB15`)^2, na.rm=T) / sum(!is.na((gt$`r1022_FDPL190729849-1a_HTWF3DSXX_L2_B73v4` - gt$`IBB15`)^2), na.rm=T) * 100
sum((gt$`r1022_FDPL190729849-1a_HTWF3DSXX_L2_B73v4` - gt$`r1049_FDPL190729869-1a_HTWF3DSXX_L3_B73v4`)^2, na.rm=T) / sum(!is.na((gt$`r1022_FDPL190729849-1a_HTWF3DSXX_L2_B73v4` - gt$`r1049_FDPL190729869-1a_HTWF3DSXX_L3_B73v4`)^2), na.rm=T) * 100



#Proportion heterozygous
GT = extract.gt(vcf)
hets = is.het(GT, na_is_false = F)
Het = data.frame("prop_het"=colSums(hets, na.rm=T) / colSums(!is.na(hets)))
Het %<>% rownames_to_column("sample")

#Flag overly heterozygous samples
flag.samps = Het %>% dplyr::filter(prop_het > 0.05)
kk %<>% mutate(sample = Vcf.ind.names)
flag.samps = merge(flag.samps, kk[,c("sample", "Names", "unique")], by = c("sample"))
flagged_samps = flag.samps %>% dplyr::select(Names, prop_het, unique) %>% arrange(Names)

#Identify heterozygous sites
var.het = data.frame("prop_het_var"=rowSums(hets, na.rm=T) / rowSums(!is.na(hets)))
var.het %<>% rownames_to_column("var")
var.het %>% dplyr::filter(prop_het_var > 0.1) %>% nrow()




```

### Make NJ tree with duplicate fastq files labelled as red; use filenames as labels
```{r}

Vcf2 = Vcf
Vcf2$ind.names = dd$names
tt2 = stamppNeisD(Vcf2, pop = F)

plot(nj(tt2),cex=0.3, type="radial", no.margin=T, label.offset = 0.01, tip.color = dd$unique)
plot(nj(tt2),cex=0.3, type="fan", no.margin=T, label.offset = 0.01, tip.color = dd$unique)

```

### Make NJ tree with duplicate fastq files w/ sample names as labels
```{r}

Vcf3 = Vcf
Vcf3$ind.names = kk$Names
tt3 = stamppNeisD(Vcf3, pop = F)

plot(nj(tt3),cex=0.3, type="radial", no.margin=T, label.offset = 0.01, tip.color = kk$unique)
plot(nj(tt3),cex=0.3, type="fan", no.margin=T, label.offset = 0.01, tip.color = kk$unique)

#Make heatmap
heatmap.2(tt3, colsep=0, rowsep=0, trace = "none", cexRow = 0.08, margin = c(1,5), offsetCol = 0, offsetRow = 0, labCol=F, dendrogram="row", key = F, keysize=0.5, ColSideColors = kk$unique)
```
###Make NJ  tree labelled with heterotic group
```{r}

plot(nj(tt3),cex=0.3, type="fan", no.margin=T, label.offset = 0.01, tip.color = KK$Cols)
legend(0.15,0.2,legend = names, fill = cols, xpd=T, cex = 0.6)
plot(nj(tt3),cex=0.3, type="radial", no.margin=T, label.offset = 0.01, tip.color = KK$Cols)
legend(-0.34,0.72,legend = names, fill = cols, xpd=T, cex = 0.55)
```

###Make heatmaps
```{r}
heatmap.2(tt3, colsep=0, rowsep=0, trace = "none", cexRow = 0.08, margin = c(1,5), offsetCol = 0, offsetRow = 0, labCol=F, dendrogram="row", key = F, keysize=0.5, ColSideColors = KK$Cols)
```


###Make PCA plots
```{r}
#glPCA functions bark due to missing values
#To find the culplrits:
toRemove <- is.na(glMean(Vcf3, alleleAsUnit = FALSE)) # TRUE where NA
which(toRemove) # position of entirely non-typed loci
Vcf3 <- Vcf3[, !toRemove]
pca = glPcaFast(Vcf3, nf = 516)

g2 <- s.label (pca$scores, xax=1, yax=2, ppoints.col = KK$Cols, paxes.draw=T, pgrid.draw =F, plot = TRUE, label = NULL)
legend(0.15,0.2,legend = names, fill = cols, xpd=T, cex = 0.6)

pc.scores = as.data.frame(pca$scores[,c(1,2)])
pc.scores = rownames_to_column(pc.scores, "Names")
pc.scores = merge(pc.scores, KK, by = c("Names"), all.x=T)
pc.scores %<>% mutate(HeteroticGroup = case_when(is.na(Group) ~ "unknown", TRUE ~ as.character(Group)))

cols2=c("gray48", "thistle", "pink4", "green","darkolivegreen3","darkolivegreen4", "blue", "gold", "lightpink", "red", "wheat3","lightblue","turquoise4","saddlebrown", "black")
pc.scores %>% ggplot(aes(y = PC1, x = PC2, color = HeteroticGroup)) + geom_point(alpha = 0.7) + scale_color_manual(values = cols2)


```


